inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/config/CommonConfigHolder.java

```diff
@@ -0,0 +1,477 @@
+    private static volatile boolean isInit = false;
+    private Map<String, String> props;
+    // pre-read field values
+    private String clusterTag = VAL_DEF_CLUSTER_TAG;
+    private String clusterName = VAL_DEF_CLUSTER_NAME;
+    private String clusterIncharges = VAL_DEF_CLUSTER_INCHARGES;
+    private String clusterExtTag = VAL_DEF_CLUSTER_EXT_TAG;
+    private String managerType = VAL_DEF_MANAGER_TYPE;
+    private IManagerIpListParser ipListParser = null;
+    private String managerAuthSecretId = "";
+    private String managerAuthSecretKey = "";
+    private long configChkInvlMs = VAL_DEF_CONFIG_CHECK_INTERVAL_MS;
+    private boolean enableAudit = VAL_DEF_ENABLE_AUDIT;
+    private final HashSet<String> auditProxys = new HashSet<>();
+    private String auditFilePath = VAL_DEF_AUDIT_FILE_PATH;
+    private int auditMaxCacheRows = VAL_DEF_AUDIT_MAX_CACHE_ROWS;
+    private long auditFormatInvlMs = VAL_DEF_AUDIT_FORMAT_INTERVAL_MS;
+    private boolean responseAfterSave = VAL_DEF_RESPONSE_AFTER_SAVE;
+    private long maxResAfterSaveTimeout = VAL_DEF_MAX_RAS_TIMEOUT_MS;
+    private boolean noTopicAccept = VAL_DEF_NOTFOUND_TOPIC_ACCEPT;
+    private boolean enableWhiteList = VAL_DEF_ENABLE_WHITELIST;
+    private int maxBufferQueueSizeKb = VAL_DEF_MAX_BUFFERQUEUE_SIZE_KB;
+    private String eventHandler = VAL_DEF_EVENT_HANDLER;
+    private String cacheClusterSelector = VAL_DEF_CACHE_CLUSTER_SELECTOR;
+    private String proxyNodeId = VAL_DEF_PROXY_NODE_ID;
+    private String msgCompressType = VAL_DEF_MSG_COMPRESS_TYPE;
+    private int prometheusHttpPort = VAL_DEF_PROMETHEUS_HTTP_PORT;
+
+    /**
+     * get instance for common.properties config manager
+     */
+    public static CommonConfigHolder getInstance() {
+        if (isInit && instance != null) {
+            return instance;
+        }
+        synchronized (CommonConfigHolder.class) {
+            if (!isInit) {
+                instance = new CommonConfigHolder();
+                if (instance.loadConfigFile()) {
+                    instance.preReadFields();
+                }
+                isInit = true;
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Get the original attribute map
+     *
+     * Notice: only the non-pre-read fields need to be searched from the attribute map,
+     *         the pre-read fields MUST be got according to the methods in the class.
+     */
+    public Map<String, String> getProperties() {
+        return this.props;
+    }
+
+    /**
+     * getStringFromContext
+     *
+     * @param context
+     * @param key
+     * @param defaultValue
+     * @return
+     */
+    public static String getStringFromContext(Context context, String key, String defaultValue) {
+        String value = context.getString(key);
+        value = (value != null) ? value : getInstance().getProperties().getOrDefault(key, defaultValue);
+        return value;
+    }
+
+    public String getClusterTag() {
```

Comment:
Too much getters and setters, can we use the `lombok` here to reduce them?

Reply:
This part is pre-reading processing, used for speed-up, and the developed code cannot be reduced through lombok